### PR TITLE
MAP-163 new view for reception areas

### PR DESCRIPTION
**Pull request by GurnankCheema**
new view and controller created to enable movement of prisoners to reception. The page is not accessible in the current flow but can be accessed by directly entering the url  (/cell-move/select-reception).
The /cell-move/select-cell page was used as a basis for the new page and the exact content isn't known yet as it is work in progress. Bare-bones tests added.

### CODE REVIEW

File: backend/controllers/cellMove/selectReception.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  const occupantAssessments = await prisonApi.getCsraAssessments(res.locals, occupantOffenderNos)
+  const assessmentsGroupedByOffenderNo = occupantAssessments ? groupBy(occupantAssessments, 'offenderNo') : []
+
+  const cellSharingAssessments = Object.keys(assessmentsGroupedByOffenderNo)
+    .map(
+      (offenderNumber) =>
+        assessmentsGroupedByOffenderNo[offenderNumber]
+          .filter(
+            (assessment) =>
+              assessment && assessment.assessmentDescription && assessment.assessmentDescription.includes('CSR')
```
Comment (steverendell) [inline backend/controllers/cellMove/selectReception.ts:54]: you can just use assessment?.assessmentDescription?.includes('CSR') here ?
(And there are several other similar situations in this file)

File: views/cellMove/selectReception.njk [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          <th scope="col" class="govuk-table__header">Relevant alerts</th>
+          <th scope="col" class="govuk-table__header"><span class="govuk-visually-hidden">Select cell</span></th>
+        </tr>
+      </thead>
+      <tbody class="govuk-table__body">
+        {% for cell in cells %}
+          {% if cell.occupants.length > 0 %}
+              {% for occupier in cell.occupants %}
+                <tr class="govuk-table__row">
+                  {{ makeRow(cell, loop.index, cell.occupants.length === loop.index, occupier) }}
```
Comment (steverendell) [inline views/cellMove/selectReception.njk:132]: there is a loop.last boolean you could use

File: backend/controllers/cellMove/selectReception.ts (hunk 1 above)
Comment (GurnankCheema) [reply backend/controllers/cellMove/selectReception.ts:54]: agreed

File: views/cellMove/selectReception.njk (hunk 2 above)
Comment (GurnankCheema) [reply views/cellMove/selectReception.njk:132]: at this stage, the display of reception on this page was merely a proof of concept. So it's not really necessary to change the logic too much when compared with the nunjucks template this page was cloned from.

Comment (steverendell) [reply views/cellMove/selectReception.njk:132]: not sure what you mean, i was just suggesting changing 'cell.occupants.length === loop.index' to the simpler but identical 'loop.last'

Comment (GurnankCheema) [reply views/cellMove/selectReception.njk:132]: oh I see

File: backend/controllers/cellMove/selectReception.ts [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    offenderNumbers: occupantOffenderNos,
+  })
+  const occupantAssessments = await prisonApi.getCsraAssessments(res.locals, occupantOffenderNos)
+  const assessmentsGroupedByOffenderNo = occupantAssessments ? groupBy(occupantAssessments, 'offenderNo') : []
+
+  const cellSharingAssessments = Object.keys(assessmentsGroupedByOffenderNo)
+    .map(
+      (offenderNumber) =>
+        assessmentsGroupedByOffenderNo[offenderNumber]
+          .filter((assessment) => assessment?.assessmentDescription.includes('CSR'))
```
Comment (steverendell) [inline backend/controllers/cellMove/selectReception.ts:52]: this is not quite the same as previously, i.e. do you know that assessmentDescription cant be null?
Also I was suggesting you could make a similar change e.g. at lines 75, 92, 104

**Pull request merged**